### PR TITLE
Store test connection output

### PR DESCRIPTION
**Pull request by pmbrull**
### Describe your changes :

So that we can pick it up in argo templates

#
### Type of change :

- [x] Bug fix
- [x] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

**Review by OnkarVO7 — APPROVED** [review]

File: ingestion/operators/docker/test_connection.py [hunk 1]
```diff
@@ -54,7 +57,10 @@ def main():
     test_connection_fn = get_test_connection_fn(
         test_service_connection.connection.config
     )
-    test_connection_fn(connection)
+    results: Optional[TestConnectionResult] = test_connection_fn(connection)
+    if results:
+        with open("/tmp/test_connection_results.json", "w", encoding=UTF_8) as file:
```
Comment (harshach) [inline ingestion/operators/docker/test_connection.py:62]: if there are multiple test connections going on , wouldn't this override the file with different connection results.
lets say there are 2 users logged in and each with their own sessions and simultaneously adding services can get unwanted results.
Also the case where we write a test_connection_results.json from previous test connection, current test connection fails , the polling of this file can return a stale result from previous file
IMO, storing these results in a file doesn't look like a good idea .
Might as well expand the connection on the server side to have test connection and write the results back there instead in a file

**Review by harshach — CHANGES_REQUESTED** [review]
Lets look at alternate options instead of writing to a file

File: ingestion/operators/docker/test_connection.py (hunk 1 above)
Comment (pmbrull) [reply ingestion/operators/docker/test_connection.py:62]: these will happen running on an ephemeral pod. When we call the test_connection from Argo, the workflow will execute on a pod, write locally, and then expose the results via the output param read from the file: https://argoproj.github.io/argo-workflows/walk-through/output-parameters/

**Comment by pmbrull** [comment]
Closing this one. When we add the status as part of the service, we won't need this to send the data out of the pod. We will communicate entirely through API: Pod -> OM API <- UI

**Comment by pmbrull** [comment]
Closing this one. When we add the status as part of the service, we won't need this to send the data out of the pod. We will communicate entirely through API: Pod -> OM API <- UI

**Pull request closed without merging**